### PR TITLE
Infrastructure: CI and hook to check coverage file

### LINKED ISSUE #2061: Add git hook and CI for updating the coverage and quality report page
Similar with what is done for `example/index.html`, this file should be generated if any related files are touched. A diff check should also be done if something is skipped locally with a check in CI.

- [ ] Identify the files that should trigger a generation/CI check
- [ ] Create script in `package.json` to call `node scripts/coverage-report.js`
- [ ] Add git hook similar to https://github.com/w3c/aria-practices/blob/45e9a6c4b19ea03356a9aa1cda8c8eaaf3272015/package.json#L73-L76
- [ ] Add new CI job similar to https://github.com/w3c/aria-practices/blob/main/.github/workflows/examples.yml, but the existing `coverage-report.yml` is doing a different job right now

### PR DESCRIPTION
**Pull request by nschonni**
Closes #2061
___
[WAI Preview Link](https://deploy-preview-266--aria-practices.netlify.app/ARIA/apg) _(Last built on Tue, 13 Feb 2024 19:23:17 GMT)._

### CODE REVIEW

**Comment by nschonni** [comment]
@jongund I think you originally set this up, so hope this makes sense

**Review by jongund — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -74,7 +75,9 @@
     ],
     "examples/**/*.html": [
       "npm run reference-tables",
-      "git add examples/index.html"
+      "git add examples/index.html",
+      "npm run coverage-report",
+      "git add coverage/"
```
Comment (nschonni) [inline package.json:80]: @howard-e are these explicit `git add` still ok with the new lint-staged?

Comment (howard-e) [reply package.json:80]: Those should be fine

**Comment by a11ydoer** [comment]
@evmiguel will faciliate the communication.

**Review by howard-e — APPROVED** [review]

**Comment by mcking65** [comment]
I've not looked at this PR for a long time; somehow I assumed that the coverage and quality report was already getting auto-updated with merges to main. Now, I'm not sure that it is. If not, we need to update this PR.

The coverage and quality report is now part of the APG. It is published in the about section on the page [Coverage and Quality Reports | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/about/coverage-and-quality/), which comes from content/about/coverage-and-quality/coverage-and-quality-report.html.

`scripts/coverage-report.js` is already updated to generate the file to the correct location.

I'd like one more change to the generated content. That is to add the date the reports were last generated under the heading "About These Reports". The first paragraph under the heading could be:

>These reports were updated on monthName, dayNumber, yearNumber.

**Comment by mcking65** [comment]
@nschonni @howard-e 

I think I have this PR updated. However, I don't understand the point of this check that is failing:

>**Ensure no git changes 0s**

If this check testing to see if a PR would change the report?

I'd assume that that many if not most PRs will change the report. That is the reason for generating a new report on each commit to main.

**Comment by nschonni** [comment]
Gave this a rebase and regenerated the files by running the `coverage-report` script

**Review by howard-e — APPROVED** [review]
@nschonni thanks. Looks good to me!

**Comment by ccanash** [comment]
Hi @mcking65  flagging this for your review

**Comment by ariellalgilmore** [comment]
I updated the `coverage-and-quality` template and script. The deploy preview won't show the `Page last updated: ...` until the PR gets merged, which will then run the `coverage-and-quality` script

**Comment by mcking65** [comment]
I manually ran the report and pushed the change so people can preview it with the new update date functionality.

[Preview the coverage and quality report page](https://deploy-preview-266--aria-practices.netlify.app/aria/apg/about/coverage-and-quality/)

File: scripts/coverage-report.js [hunk 2]
```diff
@@ -1138,12 +1138,26 @@ $('#example_summary_prototype').html(countPrototype);
...
 // cheerio seems to fold the doctype lines despite the template
 const result = $.html()
   .replace('<!DOCTYPE html>', '<!DOCTYPE html>\n')
   .replace(
     '<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">',
     '<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">\n'
+  )
+  .replace(
+    '<p>Page last updated: </p>',
+    `<p> Page last updated: ${formattedDate}</p>`
```
Comment (ariellalgilmore) [inline scripts/coverage-report.js:1160]:
```suggestion
    `<p>Page last updated: ${formattedDate}</p>`
```

**Review by mcking65 — APPROVED** [review]

File: scripts/coverage-report.template [hunk 3]
```diff
@@ -17,6 +17,7 @@
 <body>
   <main>
     <h1>Coverage and Quality Reports</h1>
+    <p>Page last updated: </p>
```
Comment (nschonni) [inline scripts/coverage-report.template:20]: As shown on the landed commit, adding a stamp will continually fail the diff, except for things landed on the same day as the last time this was updated

**Pull request merged**